src/handlers/common_key_events.rs

```diff
@@ -133,6 +133,141 @@ pub fn handle_left_event(app: &mut App) {
     app.set_current_route_state(Some(ActiveBlock::Empty), Some(ActiveBlock::Library));
 }
 
+pub fn go_first_line(app: &mut App) {
+    match app.get_current_route().active_block {
+        ActiveBlock::AlbumList => {
+            app.album_list_index = 0;
+        }
+        ActiveBlock::TrackTable => {
+            app.track_table.selected_index = 0;
+        }
+        ActiveBlock::AlbumTracks => match app.album_table_context {
+            AlbumTableContext::Full => {
+                app.saved_album_tracks_index = 0;
+            }
+            AlbumTableContext::Simplified => {
+                if let Some(selected_album) = &mut app.selected_album {
+                    selected_album.selected_index = 0;
+                }
+            }
+        },
+        ActiveBlock::Artists => {
+            app.artists_list_index = 0;
+        }
+        ActiveBlock::Artist => {
+            if let Some(artist_albums) = &mut app.artist_albums {
+                artist_albums.selected_index = 0;
+            }
+        }
+        ActiveBlock::RecentlyPlayed => {
+            app.recently_played.index = 0;
+        }
+        ActiveBlock::SearchResultBlock => match app.search_results.selected_block {
+            SearchResultBlock::AlbumSearch => {
+                app.search_results.selected_album_index = Some(0);
+            }
+            SearchResultBlock::SongSearch => {
+                app.search_results.selected_tracks_index = Some(0);
+            }
+            SearchResultBlock::ArtistSearch => {
+                app.search_results.selected_artists_index = Some(0);
+            }
+            SearchResultBlock::PlaylistSearch => {
+                app.search_results.selected_playlists_index = Some(0);
+            }
+            SearchResultBlock::Empty => {}
+        },
+        ActiveBlock::MyPlaylists => {
+            app.selected_playlist_index = Some(0);
+        }
+        _ => {}
+    }
+}
+
+pub fn go_last_line(app: &mut App) {
```

Comment:
Eventually, we could probably improve things by having a more consistent way of managing the selection index. I'll look into this as part of working on better pagination.